tests/federation_upload_keys_test.go

```diff
@@ -80,13 +80,17 @@ func TestFederationKeyUploadQuery(t *testing.T) {
 
 	// sytest: Can query remote device keys using POST
 	t.Run("Can query remote device keys using POST", func(t *testing.T) {
+		// Device list changes only come down incremental syncs, so we do an
+		// initial sync up front.
+		_, nextBatch := bob.MustSync(t, client.SyncReq{})
```

Comment:
This seems reasonable---but how did the test ever pass without this?

Reply:
I think that things were slow enough that the device list update wouldn't make it down the first `/sync`?